Source/WTF/wtf/win/OSAllocatorWin.cpp

```diff
@@ -97,19 +97,21 @@ void OSAllocator::commit(void* address, size_t bytes, bool writable, bool execut
 
 void OSAllocator::decommit(void* address, size_t bytes)
 {
-    // According to http://msdn.microsoft.com/en-us/library/aa366892(VS.85).aspx,
-    // bytes (i.e. dwSize) being 0 when dwFreeType is MEM_DECOMMIT means that we'll
-    // decommit the entire region allocated by VirtualAlloc() instead of decommitting
-    // nothing as we would expect. Hence, we should check if bytes is 0 and handle it
-    // appropriately before calling VirtualFree().
-    // See: https://bugs.webkit.org/show_bug.cgi?id=121972.
+    // https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc
+    // Use MEM_RESET to purge physical pages at timing of OS's preference. This is aligned to
+    // madvise MADV_FREE / MADV_FREE_REUSABLE.
+    // https://devblogs.microsoft.com/oldnewthing/20170113-00/?p=95185
+    // > The fact that MEM_RESET does not remove the page from the working set is not actually mentioned
+    // > in the documentation for the MEM_RESET flag. Instead, it’s mentioned in the documentation for
+    // > the Offer­Virtual­Memory function, and in a sort of backhanded w
+    // So, we need VirtualUnlock call.
     if (!bytes)
         return;
-    // Silence warning about using MEM_DECOMMIT instead of MEM_RELEASE:
-#pragma warning(suppress: 6250)
-    bool result = VirtualFree(address, bytes, MEM_DECOMMIT);
+    void* result = VirtualAlloc(address, bytes, MEM_RESET, PAGE_READWRITE);
```

Comment:
I think this is a behavior change. committing again by using `OSAllocator::commit` returned not zero-filled region. Is it OK?

Reply:
Yes, it works. And the other platforms' OSAllocator does not guarantee that zero-filled pages are produced from commit. So, the behavior is more aligned to the other platforms' one.